### PR TITLE
State management 

### PR DESCRIPTION
**Pull request by Xstoudi**
The state management is greatly inspired by NMRium way.

Three states are splitted by how they will be stored/handled and the rate of access we need on them:
* `Data` containing current work
* `Preferences` showing display preferences
* `View` containing current view

The states are modified using dispatch stored in a DispatchContext.

Example data were was added as a `Counter` component, counter data value and counter action.

### CODE REVIEW

File: src/hooks/useData.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import { DataContext } from '../components/context/DataContext';
+import { DispatchContext } from '../components/context/DispatchContext';
+
+export default function useData() {
+  return {
+    data: useContext(DataContext),
+    dispatch: useContext(DispatchContext).data,
+  };
+}
```
Comment (targos) [inline src/hooks/useData.ts:11]: Not sure about this, for two reasons:
- We will probably use multiple dispatch functions from some components. Then `dispatch` becomes ambiguous and we will need to rename it in some places (inconsistent)
- The dispatch function is stable, but by getting it from the `useData` hook, we force components that only need it to be re-rendered when data changes.

I suggest we export two hooks instead: `useData` and `useDataDispatch`.

Comment (Xstoudi) [reply src/hooks/useData.ts:11]: Make sense. Is the location of the file okay to you or would you prefer it in the state subfolder?

Comment (targos) [reply src/hooks/useData.ts:11]: location sgtm

**Comment by Xstoudi** [comment]
Manually merged

**Pull request closed without merging**